CommonLibSF/src/SFSE/Logger.cpp

```diff
@@ -21,5 +24,31 @@ namespace SFSE
 			path /= "My Games\\Starfield\\SFSE\\Logs";
 			return path;
 		}
+
+		void init()
+		{
+			auto path = log_directory();
+			if (!path)
+				return;
+			
+			const auto data = PluginVersionData::GetSingleton();
+			*path /= std::format("{}.log", data->GetPluginName());
+
+			std::vector<spdlog::sink_ptr> sinks{
+				std::make_shared<spdlog::sinks::basic_file_sink_mt>(path->string(), true), 
+				std::make_shared<spdlog::sinks::msvc_sink_mt>()
+			};
+
+			auto logger = std::make_shared<spdlog::logger>("global", sinks.begin(), sinks.end());
+#ifndef NDEBUG
+			logger->set_level(spdlog::level::debug);
+			logger->flush_on(spdlog::level::debug);
+#else
+			logger->set_level(spdlog::level::info);
+			logger->flush_on(spdlog::level::info);
+#endif
+			spdlog::set_default_logger(std::move(logger));
+			spdlog::set_pattern("[%^%L%$] %v");
```

Comment:
`[%T.%e] [%L] [%=5t] %v` maybe make this the default so it logs the time and thread id by default as well?

Reply:
I chose this specifically because it is minimal, and if you need that much info, you can still set the pattern yourself.

Reply:
Not saying I'm against it, but do those types change width at all? Because I always disliked when things don't line up.

Reply:
Sorry just saw this, yes the width is consistent (it should handle <5 digit thread ids properly)